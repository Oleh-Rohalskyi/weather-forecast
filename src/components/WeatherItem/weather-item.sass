@import values

.weather-item
    width: 325px
    height: 143px
    background-color: #FFFFFF
    border: 1px solid #17475B
    border-radius: 15px
    padding: 21px
    margin: 25px auto 22px auto
    &__info
        font-family: 'Roboto'
        color: #F3997B
        font-size: 12px
        display: flex
        & span:first-child 
            margin-right: auto
        & span:last-child
            margin-left: auto 
    &__data
        font-family: 'Roboto'
        color: #114358
        font-size: 16px
        margin-top: 7px
        margin-bottom: 20px
        line-height: 30px
    &__chip
        display: flex
        &>div
            border-radius: 11px
            background-color: #E6F4F1
            line-height: 21px
            text-align: center
            padding: 4px 19px
            height: 21px
    
