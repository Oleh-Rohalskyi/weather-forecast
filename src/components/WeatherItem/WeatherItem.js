import React from 'react';
import './weather-item.sass';

const WeatherItem = ({props}) => {

  let {temp,feels_like,description,dt_txt} = props;
  let [date,time] = dt_txt.split(" ");

  return (
    <li className="weather-item">
      <div className="weather-item__info">
        <span>{date}</span>
        <span>{time}</span>
      </div>
      <div className="weather-item__data">
        <p>temperature: {temp}</p>
        <p>feels like temperature: {feels_like}</p>
      </div>
      <div className="weather-item__chip">{description}</div>
    </li>
  );
};

export default WeatherItem;
