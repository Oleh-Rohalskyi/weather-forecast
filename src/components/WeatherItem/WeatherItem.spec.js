import React from 'react';
import { mount } from 'enzyme';
import WeatherItem from './WeatherItem';


describe('src > components > WeatherItem', () => {

  const mockUp = {
    temp: "36.6",
    feels_like: "42",
    description: "Thunderly",
    dt_txt: "2020-02-15 12:00:00"
  }

  it('renders without crashing', () => {

    const wrapper = mount(
      <WeatherItem key={0} props={mockUp} />
    );
    expect(wrapper).toMatchSnapshot();

  });

});
