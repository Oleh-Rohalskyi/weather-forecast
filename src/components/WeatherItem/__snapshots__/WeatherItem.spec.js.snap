// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src > components > WeatherItem renders without crashing 1`] = `
<WeatherItem
  props={
    Object {
      "description": "Thunderly",
      "dt_txt": "2020-02-15 12:00:00",
      "feels_like": "42",
      "temp": "36.6",
    }
  }
>
  <li
    className="weather-item"
  >
    <div
      className="weather-item__info"
    >
      <span>
        2020-02-15
      </span>
      <span>
        12:00:00
      </span>
    </div>
    <div
      className="weather-item__data"
    >
      <p>
        temperature: 
        36.6
      </p>
      <p>
        feels like temperature: 
        42
      </p>
    </div>
    <div
      className="weather-item__chip"
    >
      Thunderly
    </div>
  </li>
</WeatherItem>
`;
